agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -201,6 +201,7 @@ def start(info)
         end
       end
 
+      attach_router unless interface_ip('weave')
```

Comment:
I suggest fixing the `interface_ip()` `rescue Errno::EADDRNOTAVAIL` `return nil` behavior and then LGTM